web/src/components/AddressAutocomplete/AddressAutocomplete.js

```diff
@@ -0,0 +1,160 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { TextField, Label } from '@redwoodjs/forms'
+
+const AddressAutocomplete = (props) => {
+  const address1FieldRef = useRef(null)
+  const [addressStreet, setAddressStreet] = useState(props.addressStreet || '')
+  const [postcode, setPostcode] = useState('')
+  const [statelocation, setStatelocation] = useState('')
+  const [localCity, setLocalCity] = useState('')
+  const [country, setCountry] = useState('')
+  const [addressUnitNumber] = useState('')
+
+  useEffect(() => {
+    const handleScriptLoad = (address1FieldRef) => {
+      const autoComplete = new window.google.maps.places.Autocomplete(
+        address1FieldRef.current,
+        { fields: ['address_components', 'geometry'], types: ['address'] }
+      )
+
+      autoComplete.setFields(['address_components', 'formatted_address'])
+      autoComplete.addListener('place_changed', () =>
+        handlePlaceSelect(autoComplete)
+      )
+    }
+    loadScript(
+      `https://maps.googleapis.com/maps/api/js?key=${process.env.GOOGLE_MAP_API_KEY}&libraries=places`,
+      () => handleScriptLoad(address1FieldRef)
+    )
+  }, [])
+
+  const loadScript = (url, callback) => {
+    let script = document.createElement('script')
```

Comment:
```suggestion
    const script = document.createElement('script')
```

Reply:
Can we add an `id` to this script? Then we should check if the script already exists before trying to append it to the DOM.

Reply:
added   
`script.id = 'data'
    if (document.getElementById('data')) {
      return
    }`